readap-js/src/client.ts

```diff
@@ -0,0 +1,299 @@
+import init, { 
+  UrlBuilder, 
+  parse_dds, 
+  parse_das, 
+  parse_dods,
+  type DodsData
+} from 'readap-wasm';
+
+import type { 
+  DAPClientOptions, 
+  VariableConstraints, 
+  FetchDataOptions,
+  DatasetInfo,
+  VariableData,
+  DdsDataset,
+  DasAttributes,
+  VariableInfo,
+  CoordinateInfo
+} from './types.js';
+
+export class DAPClient {
+  private baseUrl: string;
+  private options: DAPClientOptions;
+  private urlBuilder?: UrlBuilder;
+  private ddsCache?: DdsDataset;
+  private dasCache?: DasAttributes;
+  private initialized = false;
+
+  constructor(baseUrl: string, options: DAPClientOptions = {}) {
+    this.baseUrl = baseUrl.replace(/\/$/, ''); // Remove trailing slash
+    this.options = {
+      timeout: 30000,
+      ...options
+    };
+  }
+
+  async init(): Promise<void> {
+    if (this.initialized) return;
+    
+    // In Node.js, we need to load the WASM file manually
+    if (typeof window === 'undefined') {
+      const fs = await import('fs');
+      const path = await import('path');
+      const { fileURLToPath } = await import('url');
+      
+      // Find the WASM file relative to this module
+      const currentDir = path.dirname(fileURLToPath(import.meta.url));
+      const wasmPath = path.resolve(currentDir, '../../readap-wasm/pkg/readap_wasm_bg.wasm');
```

Comment:
This won't work in a production setting. Instead use the workspace build commands to copy the wasm file from its build location into this packages diet so it's always available at runtime. Do the same for all the ts and js files needed as well.